src/ngraph/op/convolution.cpp

```diff
@@ -25,16 +25,476 @@
+            m_pads_begin.clear();
+            m_pads_end.clear();
+            auto filter_shape = filters_shape.to_shape();
+            filter_shape.erase(filter_shape.begin(), filter_shape.begin() + 2); // Remove {O,I}
+            infer_auto_padding(data_batch_shape.to_shape(),
+                               filter_shape,
+                               m_strides,
+                               m_dilations,
+                               m_auto_pad,
+                               m_pads_end,
+                               m_pads_begin);
+        }
+    }
+
+    element::Type result_et;
+    PartialShape result_shape;
+
+    NODE_VALIDATION_CHECK(
+        this,
+        element::Type::merge(result_et, data_batch_et, filters_et),
+        "Element types for data batch and filters do not match (data batch element type: ",
+        data_batch_et,
+        ", filters element type: ",
+        filters_et,
+        ").");
+
+    result_shape =
+        infer_convolution_forward(this,
+                                  data_batch_shape,
+                                  Strides(static_cast<size_t>(data_batch_shape.rank()) - 2, 0),
+                                  m_pads_begin,
+                                  m_pads_end,
+                                  filters_shape,
+                                  m_strides,
+                                  m_dilations);
+
+    set_output_type(0, result_et, result_shape);
+}
+
+shared_ptr<Node> op::v1::Convolution::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<v1::Convolution>(new_args.at(0),
+                                        new_args.at(1),
+                                        m_strides,
+                                        m_pads_begin,
+                                        m_pads_end,
+                                        m_dilations,
+                                        m_auto_pad);
+}
+
+void op::v1::Convolution::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    auto delta = deltas.at(0);
+
+    auto x = input_value(0);
+    const auto x_shape = x.get_shape();
+
+    auto f = input_value(1);
+    const auto f_shape = f.get_shape();
+
+    adjoints.add_delta(x,
+                       make_shared<op::v1::ConvolutionBackpropData>(
+                           x_shape, f, delta, m_strides, m_dilations, m_pads_begin, m_pads_end));
+
+    adjoints.add_delta(f,
+                       make_shared<op::v1::ConvolutionBackpropFilters>(
+                           x, f_shape, delta, m_strides, m_dilations, m_pads_begin, m_pads_end));
+}
+
+constexpr NodeTypeInfo op::v1::ConvolutionBackpropData::type_info;
+shared_ptr<Node> op::v1::Convolution::get_default_value() const
+{
+    return ngraph::make_constant_from_string("0", get_element_type(), get_shape());
+}
+
+op::v1::ConvolutionBackpropData::ConvolutionBackpropData(const Shape& data_batch_shape,
```

Comment:
At some point in the near future we are going to need to make this dynamic, but let's stick with this for this PR.